src/m365/spo/commands/tenant/tenant-appcatalog-add.ts

```diff
@@ -97,72 +97,77 @@ class SpoTenantAppCatalogAddCommand extends SpoCommand {
-    await this.createAppCatalog(args.options, logger);  
+    await this.createAppCatalog(args.options, logger);
   }
 
-  private ensureNoExistingSite(url: string, force: boolean, logger: Logger): Promise<void> {
-    return new Promise<void>((resolve: () => void, reject: (err: CommandError) => void): void => {
+  private async ensureNoExistingSite(url: string, force: boolean, logger: Logger): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Checking if site ${url} exists...`);
+    }
+
+    const siteGetOptions = {
+      options: {
+        url: url,
+        verbose: this.verbose,
+        debug: this.debug,
+        _: []
+      }
+    };
+
+    try {
+      await Cli.executeCommandWithOutput(spoSiteGetCommand as Command, siteGetOptions);
+
       if (this.verbose) {
-        logger.logToStderr(`Checking if site ${url} exists...`);
+        logger.logToStderr(`Found site ${url}`);
       }
 
-      const siteGetOptions = {
-        options: {
-          url: url,
-          verbose: this.verbose,
-          debug: this.debug,
-          _: []
-        }
+      if (!force) {
+        throw new CommandError(`Another site exists at ${url}`);
+      }
+
+      if (this.verbose) {
+        logger.logToStderr(`Deleting site ${url}...`);
+      }
+
+      const siteRemoveOptions = {
+        url: url,
+        skipRecycleBin: true,
+        wait: true,
+        confirm: true,
+        verbose: this.verbose,
+        debug: this.debug
       };
-      Cli
-        .executeCommandWithOutput(spoSiteGetCommand as Command, siteGetOptions)
-        .then(() => {
-          if (this.verbose) {
-            logger.logToStderr(`Found site ${url}`);
-          }
-
-          if (!force) {
-            return reject(new CommandError(`Another site exists at ${url}`));
-          }
-
-          if (this.verbose) {
-            logger.logToStderr(`Deleting site ${url}...`);
-          }
-          const siteRemoveOptions = {
-            url: url,
-            skipRecycleBin: true,
-            wait: true,
-            confirm: true,
-            verbose: this.verbose,
-            debug: this.debug
-          };
-          Cli
-            .executeCommand(spoSiteRemoveCommand as Command, { options: { ...siteRemoveOptions, _: [] } })
-            .then(() => resolve(), (err: CommandError) => reject(err));
-        }, (err: CommandErrorWithOutput) => {
-          if (err.error.message !== 'File Not Found.' && err.error.message !== '404 FILE NOT FOUND') {
-            // some other error occurred
-            return reject(err.error);
-          }
-
-          if (this.verbose) {
-            logger.logToStderr(`No site found at ${url}`);
-          }
-
-          // site not found. continue
-          resolve();
-        });
-    });
+
+
+      await Cli.executeCommand(spoSiteRemoveCommand as Command, { options: { ...siteRemoveOptions, _: [] } });
+    }
+    catch (err: any) {
+      if (err instanceof CommandErrorWithOutput) {
```

Comment:
I still think this can be done shorter. What do you think about the following if statement? 

```typescript
if (err.error?.message !== 'File not Found' && err.error?.message !== '404 FILE NOT FOUND') {
    throw err.error || err;
}
```

Reply:
That's way better, thank you Martin! 🙏